### DISCUSSION TITLE
Reading input files from a list

### DISCUSSION THREAD

**Discussion by dimk1** [open-webui/open-webui#7790] [General]
Hello,

I'm using Webui 0.4.8 with the latest Ollama3.3 installed with Docker and currently running it in on an AWS EC2 instance.

I attach to the chat 4 files (not via the Knowledge functionality), let's call them `FILE_1.md, FILE_2.md, FILE_3.md, FILE_4.md`. I have setup my prompt as follows:

```
### Variables
**Filenames:** `FILES = ["FILE_1.md", "FILE_2.md", "FILE_3.md", "FILE_4.md"]`

### Task:
Evaluate each file with a filename listed in the list `FILES` against criterion X.

### Output Format for Batch Evaluations: 
For all files with filenames in the list `PROPOSALS` do X & Y

```
Prompts like this always fail and the system complains that there are no files attached to work with, so it improvises. When I modify the prompt to work on a single file it works as expected. Is there any suggested format when writing prompts to declare lists with filenames and then instruct the model to do certain actions on each file recursively?

**Comment by spammenotinoz** [comment]
I don't think LLMs understand a the concept of files\attachments the way we do.  It's all just a lot of tokens in.
You can kind of get around this some creative system prompts  to treat learned knowledge as files.
